src/kintone/client.ts

```diff
@@ -63,33 +69,11 @@ const buildHttpsAgent = (options: {
     return new https.Agent({ ...clientAuth });
   }
 
-  const { protocol, hostname, port, username, password } = new URL(
-    options.proxy,
-  );
-  const proxyOptions: httpsProxyAgent.HttpsProxyAgentOptions = {
-    protocol: protocol,
-    host: hostname,
-    port: port,
+  const proxyOptions: HttpsProxyAgentOptions<ProxyAgentOptions> = {
```

Comment:
Specifying `ProxyAgentOptions` to `HttpsProxyAgentOptions` seems to be incorrect.
The type variable `T` of `HttpsProxyAgentOptions` is for specifying the protocol.
According to the type definition, the type variable `T` is passed to `ConnectOpts` and then `Protocol`.
It is used to determine the input URI is HTTP or HTTPS.
https://github.com/TooTallNate/proxy-agents/blob/e625d10c7f7b018713916f70be9f2055c5db3542/packages/https-proxy-agent/src/index.ts#L26

It's better to use the type with string (`HttpsProxyAgentOptions<string>`) or the parameter type of the constructor (`ConstructorParameters<typeof HttpsProxyAgent>[1]`).